File: client/python/mesh/prsim/licenser.py
#
# Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
# TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
#
#
from abc import ABC, abstractmethod
from typing import Dict

from mesh.kinds import License
from mesh.macro import mpi


class Licenser(ABC):

    @abstractmethod
    @mpi(name="mesh.license.imports", flags=2)
    def imports(self, text: str):
        """
        Import the license.
        :param text:
        :return:
        """
        pass

    @abstractmethod
    @mpi(name="mesh.license.exports", flags=2)
    def exports(self) -> str:
        """
        Exports the license.
        :return:
        """
        pass

    @abstractmethod
    @mpi(name="mesh.license.explain", flags=2)
    def explain(self) -> License:
        """
        Explain the license.
        :return:
        """
        pass

    @abstractmethod
    @mpi(name="mesh.license.verify", flags=2)
    def verify(self) -> bool:
        """
        Verify the license.
        :return:
        """
        pass

    @abstractmethod
    @mpi(name="mesh.license.features", flags=2)
    def features(self) -> Dict[str, str]:
        """
        License features.
        :return:
        """
        pass

File: client/java/src/main/java/io/be/mesh/prsim/Licenser.java
/*
 * Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
 * TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 *
 *
 */
package io.be.mesh.prsim;

import io.be.mesh.macro.MPI;
import io.be.mesh.macro.SPI;
import io.be.mesh.struct.License;

import java.util.Map;

/**
 * @author coyzeng@gmail.com
 */
@SPI("mesh")
public interface Licenser {

    /**
     * Import the license.
     */
    @MPI(value = "mesh.license.imports", flags = 2)
    void imports(String text);

    /***
     * Exports the license.
     */
    @MPI(value = "mesh.license.exports", flags = 2)
    String exports();

    /**
     * Explain the license.
     */
    @MPI(value = "mesh.license.explain", flags = 2)
    License explain();

    /**
     * Verify the license.
     */
    @MPI(value = "mesh.license.verify", flags = 2)
    long verify();

    /**
     * License features.
     */
    @MPI(value = "mesh.license.features", flags = 2)
    Map<String, String> features();
}
